packages/forms/resources/views/components/repeater/table.blade.php

```diff
@@ -170,10 +170,15 @@ class="fi-fo-table-repeater-actions"
                                                     $wire,
                                                 })"
                                                 @if ($afterStateUpdatedJs = $schemaComponent->getAfterStateUpdatedJs())
-                                                    x-init="{{ implode(';', array_map(
-                                                        fn (string $js): string => '$wire.watch(' . Js::from($schemaComponentStatePath) . ', ($state, $old) => ($state !== undefined) && eval(' . Js::from($js) . '))',
-                                                        $afterStateUpdatedJs,
-                                                    )) }}"
+                                                    @php
+                                                        $watchers = array_map(function (string $js) use ($schemaComponentStatePath) {
+                                                            return '$wire.watch(' . Js::from($schemaComponentStatePath) . ', (state, old) => {' .
+                                                                        'const deepEqual = (a, b) => { try { return JSON.stringify(a) === JSON.stringify(b); } catch(e) { return a === b; } };' .
```

Comment:
Can we define this new function just once instead of in every component? Then we can revert most of these changes and just add `&& (! deepEqual($state, $old))` before `eval()`

I think it belongs in `schemas/index.js` alongside `handleFormValidationError()`, and then it will be accessible in all schemas

I think it should be called `isStateEqual()`